### ISSUE TITLE
Multiline SMS input

### ISSUE THREAD

**Issue by bprfh** [andyholmes/gnome-shell-extension-gsconnect#771]
**Is your feature request related to a problem? Please describe.**
When writting sms via gsconnect, I can't make a newline.
More often than not, this means that I have to use my phone, as the message would look weird otherwise.

**Describe the solution you'd like**
It would be a really cool feature if the sms text input field would allow newlines, e.g. a textarea or multiline input.

**Describe alternatives you've considered**
Don't know of any alternatives

**System Details (please complete the following information):**
 - GSConnect version: 31
   - Installed from: extensions.gnome.org

**Additional context**
None

**Comment by andyholmes** [comment]
You can add a newline using <kbd>Shift</kbd> + <kbd>Return</kbd>, but there are no plans to change the entry widget itself at this point.

**Comment by ferdnyc** [comment]
Since commit 96d0388, <kbd>Shift</kbd>+<kbd>Enter</kbd> should insert a newline in the SMS input field. Can you check whether that works for you?
Duplicate of #464

**Comment by ferdnyc** [comment]
Whoops, too slow.

**Comment by bprfh** [comment]
Thanks for the quick Response, yes that works.
Can we add this to the wiki maybe under "common problems" or via a new page "Tipps&Tricks"?
Or was I just blind and there was a help where I should have looked?

**Comment by andyholmes** [comment]
The [Features](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Features#sms) page would probably be the logical place for this in the Wiki, but we should probably add a shortcuts help window to the application at some point.

**Comment by ferdnyc** [comment]
I'll try to get a mention of <kbd>Shift</kbd>+<kbd>Enter</kbd> into that page tonight or tomorrow. 

When I looked at it earlier, I noticed some other things out of date or missing as well. I've already updated the file manager extension section to mention Nemo in addition to Nautilus, and I'll try to fill in whatever else I spot, or as much as I can.

**Comment by ferdnyc** [comment]
1. Added the [SMS Input section](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Features#sms-input) — can anyone think of any input options besides newline entry and emoji input that should be listed there?
1. (Unrelated) Updated the instructions for enabling Legacy SMS mode, with screenshot.

**Comment by andyholmes** [comment]
> can anyone think of any input options besides newline entry and emoji input that should be listed there?

I think we have a keybinding for sending, don't we? <kbd>Ctrl</kbd>+<kbd>Return</kbd> or something?

**Comment by ferdnyc** [comment]
Nah, just plain <kbd>Enter</kbd> sends (hence <kbd>Shift</kbd>+<kbd>Enter</kbd>)... though I suppose it couldn't hurt to mention that, anyway.

**Comment by bprfh** [comment]
Sorry for the late reply, thanks for adding that.

**Comment by eeshugerman** [comment]
> there are no plans to change the entry widget itself at this point.

Do you mind if I open an issue to track interest in this? I would really appreciate the feature.

**Comment by andyholmes** [comment]
We can re-open this one, but I still don't have any plans to do this. Dealing with GtkTextView is not fun at all, and writing a custom widget even less so.

**Comment by eeshugerman** [comment]
Understood. I just think it can be good to have an issue that people can +1 if they want the feature, as an "outlet" if you will.

So let it be known: :+1:  this comment if you want a mutliline SMS input widget (or, better yet, open a PR :stuck_out_tongue_winking_eye:).

**Comment by ferdnyc** [comment]
There is, of course, the additional wrinkle that using a multiline widget would _preclude_ the current widget's <kbd>Enter</kbd>-to-send mapping, changing the established functionality of the SMS entry field.

**Comment by ferdnyc** [comment]
(Sure, with a little hoop-jumping you can probably manage a <kbd>Shift</kbd>- or <kbd>Ctrl</kbd>-<kbd>Enter</kbd> binding for newlines, and still use <kbd>Enter</kbd> to send the message. But that seems like it risks and a lot of accidental partial sends and confused users, because a multiline widget generally implies the ability to type multiple lines using <kbd>Enter</kbd>.)

**Comment by eeshugerman** [comment]
I can't speak to the technical challenges, as I've never worked with GTK, but I think ideal UX would be:
- keep <kbd>Enter</kbd> for send and <kbd>Shift</kbd>-<kbd>Enter</kbd> for newline
- text box initially has a height of one line, and only grows in height to fit the content
- text wraps
- no scrollbar until the height is 5 or 10 lines or so
- no handle to resize the textbox with the mouse

That's how [messages.google.com/web](messages.google.com/web) works, for what it's worth.

I think the second point (initial height = 1) and last point (no resize handle) go a long way toward communicating to the user that <kbd>Enter</kbd> will send the message.
